src/util/pelz_io.c

```diff
@@ -68,18 +70,20 @@ int get_file_ext(charbuf buf, int *ext)
 
 int key_load(size_t key_id_len, unsigned char *key_id, size_t * key_len, unsigned char **key)
 {
+  int return_value = 1;
   UriUriA key_id_data;
-  unsigned char tmp_key[MAX_KEY_LEN + 1];
-  FILE *key_key_f = 0;
 
   const char *error_pos = NULL;
-
   char *key_uri_to_parse = NULL;
 
   // URI parser expects a null-terminated string to parse,
   // so we embed the key_id in a 1-longer array and
   // ensure it is null terminated.
   key_uri_to_parse = (char *) calloc(key_id_len + 1, 1);
+  if (key_uri_to_parse == NULL)
+  {
+    return 1;
```

Comment:
Since you've already declared ```return_value = 1```, you should probably use it even outside the switch.